File: hash/skip_list/skip_list.py

class SkipList:  # LeetCode Q.1206.
    def __init__(self):
        self.nums2counts = dict()

    def search(self, target: int):
        if target not in self.nums2counts.keys():
            return False
        if self.nums2counts[target] == 0:
            return False

        return True

    def add(self, num: int):
        if num not in self.nums2counts.keys():
            self.nums2counts.update({num: 0})
        self.nums2counts[num] += 1

    def erase(self, num: int):
        if num not in self.nums2counts.keys():
            return False

        self.nums2counts[num] -= 1
        if self.nums2counts[num] == -1:
            self.nums2counts[num] += 1  # Go back to 0.
            return False
        return True

File: hash/skip_list/skip_list.js

var Skiplist = function () { // LeetCode Q.1206.
    this.nums2counts = new Map();
};

/** 
 * @param {number} target
 * @return {boolean}
 */
Skiplist.prototype.search = function (target) {
    if (this.nums2counts.has(target) && this.nums2counts.get(target) > 0)
        return true;
    return false;
};

/** 
 * @param {number} num
 * @return {void}
 */
Skiplist.prototype.add = function (num) {
    if (!this.nums2counts.has(num))
        this.nums2counts.set(num, 0);
    this.nums2counts.set(num, this.nums2counts.get(num) + 1);
};

/** 
 * @param {number} num
 * @return {boolean}
 */
Skiplist.prototype.erase = function (num) {
    if (this.nums2counts.has(num) && this.nums2counts.get(num) > 0) {
        this.nums2counts.set(num, this.nums2counts.get(num) - 1);
        return true;
    }
    return false;
};